src/modules/SambaAD.pm

```diff
@@ -71,12 +71,14 @@ sub IsDHCPClient {
     my ($self, $force) = @_;
 
     return $dhcp_client if (defined $dhcp_client) && !$force;
-
-    my $network         = YaPI::NETWORK->Read ();
     $dhcp_client        = TRUE;
-    foreach my $iface (values %{$network->{"interfaces"}}) {
-      $dhcp_client      = $dhcp_client && (($iface->{"bootproto"} || "") =~ m/^dhcp[46]?$/);
+    foreach my $devnum (sort keys %{LanItems->Items}){
+      LanItems->current($devnum);
+      if (LanItems->IsCurrentConfigured()){
+        $dhcp_client      = $dhcp_client && ((LanItems->bootproto || "") =~ m/^dhcp[46]?$/);
```

Comment:
hmm, ```LanItems::find_dhcp_ifaces``` is not enough?

Reply:
you'll also need to call ```LanItems::Read``` at some point, ```Items``` is not initialized otherwise.

Reply:
@mchf `LanItems::find_dhcp_ifaces` is not available through component system. Initialize is true that should be done, in my case it find that my card is dhcp configured, but if there is no card it will be also true.

Reply:
ok, then you can either make it available or at least use as inspiration what values are possible (in other words dhcp, dhcp4, dhcp6 is not everything ;-)

Reply:
@mchf well, my goal is to be identical as old code. I do not know what samba consider as good values for proposing dhcp managed samba

Reply:
And who says they did it correctly and not only just a quick hack ... ? As the last value I'm pointing to says "use dhcp or calculate a fallback when there is no dhcp server" I don't see why not to support that.

Reply:
btw if you don't want to use / enable for use ```find_dhcp_ifaces``` then you can use ```LanItems::GetDeviceMap``` which saves you setting the ```current``` crap